### PR TITLE
hot fixes for encode and decode of numeric stats mixin and intcol pro…

### PR DESCRIPTION
**Pull request by ksneab7**
…filer

### CODE REVIEW

File: dataprofiler/profilers/numerical_column_stats.py [hunk 1]
```diff
@@ -79,6 +79,8 @@ def __init__(self, options: NumericalOptions = None) -> None:
         self._mode_is_enabled: bool = True
         self.num_zeros: int = 0
         self.num_negatives: int = 0
+        self._num_quantiles: int = 1000
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:82]: nice

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:82]: we should maintain the todo for making it an option
-- or --
add it as an issue to the repo.

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:82]: or both

File: dataprofiler/tests/profilers/utils.py [hunk 2]
```diff
@@ -191,6 +191,8 @@ def assert_profiles_equal(actual, expected):
         # Condition to test whether the types are equal when a value can be float or float64
         if type(actual_value) is np.float64 or type(expected_value) is np.float64:
             assert type(float(actual_value)) == type(float(expected_value))
+        elif type(actual_value) is np.int64 or type(expected_value) is np.int64:
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/utils.py:194]: this and the above line actually feels strange. why do we have to cast and why would it be different?//

Comment (JGSweets) [reply dataprofiler/tests/profilers/utils.py:194]: what is the condition that occurs when they aren't the same?/

Comment (ksneab7) [reply dataprofiler/tests/profilers/utils.py:194]: This was my initial gripe about the types. Initialization without any processed data is float and int types.
Then when we actually process (or load in) a profile they are converted to int64s and float64s. This happens on this test: `test_json_decode`

Comment (ksneab7) [reply dataprofiler/tests/profilers/utils.py:194]: But once we do the readin for data (like in `test_json_decode_after_update`) these conditions dont really do anything

File: dataprofiler/profilers/numerical_column_stats.py [hunk 3]
```diff
@@ -79,6 +79,8 @@ def __init__(self, options: NumericalOptions = None) -> None:
         self._mode_is_enabled: bool = True
         self.num_zeros: int = 0
         self.num_negatives: int = 0
+        self._num_quantiles: int = 1000  # TODO: add to options
```
Comment (taylorfturner) [inline dataprofiler/profilers/numerical_column_stats.py:82]: good TODO here -- can you add an issue and link as well?

File: dataprofiler/profilers/numerical_column_stats.py [hunk 4]
```diff
@@ -358,7 +359,7 @@ def report(self, remove_disabled_flag: bool = False) -> dict:
 
         return profile
 
-    def _load_hist_helper(self, data):
+    def _load_stats_helper(self, data):
```
Comment (taylorfturner) [inline dataprofiler/profilers/numerical_column_stats.py:362]: just a re-name?

File: dataprofiler/profilers/numerical_column_stats.py (hunk 3 above)
Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py:82]: https://github.com/capitalone/DataProfiler/issues/853

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/numerical_column_stats.py (hunk 4 above)
Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:362]: yes since the function does a lot more now.

Comment (JGSweets) [reply dataprofiler/profilers/numerical_column_stats.py:362]: Should add a docstring though.

File: dataprofiler/profilers/numerical_column_stats.py [hunk 5]
```diff
@@ -1551,7 +1584,9 @@ def _get_quantiles(self) -> None:
 
         :return: list of quantiles
         """
-        percentiles: np.ndarray = np.linspace(0, 100, len(self.quantiles) + 2)[1:-1]
+        percentiles: np.ndarray = np.linspace(0, 100, (self._num_quantiles - 1) + 2)[
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:1587]: i like this clarity with the `()

File: dataprofiler/tests/profilers/test_int_column_profile.py [hunk 6]
```diff
@@ -1228,6 +1226,7 @@ def test_json_encode_after_update(self, time):
                     "_mode_is_enabled": True,
                     "num_zeros": 1,
                     "num_negatives": 0,
+                    "_num_quantiles": 1000,
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_int_column_profile.py:1229]: nice

File: dataprofiler/profilers/numerical_column_stats.py [hunk 7]
```diff
@@ -358,9 +359,18 @@ def report(self, remove_disabled_flag: bool = False) -> dict:
...
-        hist = data.pop("_stored_histogram")
+    def _load_stats_helper(self, deserialized_data: dict):
+        """Assistance function in the deserialization of profiler objects.
+
+            This function is to be used to enforce correct typing for attributes
+            associated with the NumericStatsMixinconversions when loading profiler
+            objects in from their serialized saved format
+        :var deserialized_data: The dictionary read in from the serialized form
+            of the profiler object
+        :type deserialized_data: Dict
```
Comment (taylorfturner) [inline dataprofiler/profilers/numerical_column_stats.py:370]:
```suggestion
        :param deserialized_data: The dictionary read in from the serialized form
            of the profiler object
        :rtype deserialized_data: Dict
```

File: dataprofiler/profilers/numerical_column_stats.py [hunk 8]
```diff
@@ -358,9 +359,18 @@ def report(self, remove_disabled_flag: bool = False) -> dict:
...
-        # Pop off stuff specific to numerical stats
-        hist = data.pop("_stored_histogram")
+    def _load_stats_helper(self, deserialized_data: dict):
+        """Assistance function in the deserialization of profiler objects.
+
+            This function is to be used to enforce correct typing for attributes
+            associated with the NumericStatsMixinconversions when loading profiler
+            objects in from their serialized saved format
+        :var deserialized_data: The dictionary read in from the serialized form
+            of the profiler object
```
Comment (taylorfturner) [inline dataprofiler/profilers/numerical_column_stats.py:369]:
```suggestion
        :param deserialized_data: The dictionary read in from the serialized form
            of the profiler object
```

**Review by taylorfturner — APPROVED** [review]
LGTM -- great work

File: dataprofiler/profilers/numerical_column_stats.py [hunk 9]
```diff
@@ -53,13 +53,13 @@ def __init__(self, options: NumericalOptions = None) -> None:
...
+        self.max: int | float | np.float64 | np.int64 | None = None
         self._top_k_modes: int = 5  # By default, return at max 5 modes
-        self.sum: int | float = 0
-        self._biased_variance: float = np.nan
-        self._biased_skewness: float | np.float64 = np.nan
-        self._biased_kurtosis: float | np.float64 = np.nan
+        self.sum: int | float | np.float64 | np.int64 = np.float64(0)
+        self._biased_variance: float | np.float64 = np.float64(np.nan)
+        self._biased_skewness: float | np.float64 = np.float64(np.nan)
+        self._biased_kurtosis: float | np.float64 = np.float64(np.nan)
```
Comment (ksneab7) [inline dataprofiler/profilers/numerical_column_stats.py:62]:
```suggestion
        self._biased_variance: float | np.float64 = np.nan
        self._biased_skewness: float | np.float64 = np.nan
        self._biased_kurtosis: float | np.float64 = np.nan
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/numerical_column_stats.py [hunk 10]
```diff
@@ -371,6 +381,30 @@ def _load_hist_helper(self, data):
                 }
             self._stored_histogram[key] = value
 
+        # Convert values to correct types
+        if self.min is not None and type(self.min) not in [np.float64, np.int64]:
+            self.min = (
+                np.float64(self.min) if type(self.min) == float else np.int64(self.min)
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:387]: Should `==` be `is`?

File: dataprofiler/profilers/numerical_column_stats.py [hunk 11]
```diff
@@ -358,9 +359,18 @@ def report(self, remove_disabled_flag: bool = False) -> dict:
...
+
+            This function is to be used to enforce correct typing for attributes
+            associated with the NumericStatsMixinconversions when loading profiler
+            objects in from their serialized saved format
+        :param deserialized_data: The dictionary read in from the serialized form
+            of the profiler object
+        :type deserialized_data: Dict
+        """
+        # Pop off attributes specific to numerical stats histogram creation
+        hist = deserialized_data.pop("_stored_histogram")
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:373]: do we even need to pass in the `deserialized_data`? are the values already assigned and can we just update those instead?

File: dataprofiler/profilers/numerical_column_stats.py [hunk 12]
```diff
@@ -371,6 +381,30 @@ def _load_hist_helper(self, data):
                 }
             self._stored_histogram[key] = value
 
+        # Convert values to correct types
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:384]: do we not need to also update: `self.histogram_methods`?

Comment (ksneab7) [reply dataprofiler/profilers/numerical_column_stats.py:384]: I dont believe so, the tests have passed with current typing.

**Review by tyfarnan — APPROVED** [review]

**Pull request merged**